src/init/initHelpers.cpp

```diff
@@ -1,20 +1,18 @@
 #include "initHelpers.hpp"
+#include <sys/resource.h>
 
 bool Init::isSudo() {
     return getuid() != geteuid() || !geteuid();
 }
 
 void Init::gainRealTime() {
-    const int                minPrio = sched_get_priority_min(SCHED_RR);
-    int old_policy;
-    struct sched_param param;
-
-    if (pthread_getschedparam(pthread_self(), &old_policy, &param)) {
-        Debug::log(WARN, "Failed to get old pthread scheduling priority");
+    // If something else is adjusting priority, don't interfere
+    if (getpriority(PRIO_PROCESS, 0) != 0) {
```

Comment:
`getpriority` checks `nice` value which is only valid for `SCHED_OTHER` (default, time-sharing). If realtime priority is set externally `nice` value may remain unchanged (zero). For example, on FreeBSD this line detects `nice Hyprland` but **not** `rtprio 31 Hyprland` and `idprio 31 Hyprland`.

Maybe instead try comparing `old_policy` and `param.sched_priority` values against defaults.

Reply:
The issue this pr is fixing is a conflict between this code setting Hyprland to realtime and external programs that try to modify the nice value e.g ananicy-cpp. 

That combination causes polkit and some other things to stop working correctly. Hence why we are looking for a chance in nice value.

The same issue shouldn't happen with externally set realtime priorities as this will just overwrite those, so I'm not sure such a change is warranted or needed.